apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerRetryHelpers.cs

```diff
@@ -0,0 +1,109 @@
+﻿// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Api.Gax;
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using static Google.Api.Gax.Grpc.RetrySettings;
+
+namespace Google.Cloud.Spanner.Data
+{
+    /// <summary>
+    /// Helper class for retrying Spanner operations.
+    /// Note that this is not the same as the retrying mechanism for each
+    /// of the RPCs, but for a higher level kind of operation that may include
+    /// one or several RPCs.
+    /// A good example of a higher level operation is the retrying of a 
+    /// transaction commit on abort, which requieres the retrying of all 
+    /// the transaction work and of the commit.
+    /// TODO(atarafamas, jskeet): Consider making public or entirely eliminating once we know 
+    /// what we can reuse from GAX retry helpers.
+    /// </summary>
+    internal static class SpannerRetryHelpers
```

Comment:
Just as a heads-up, we'll be changing some of the retry code in GAX soon. Maybe we want to bear this (and some Firestore code) in mind when doing that, to make it more amenable to reuse.